### PR TITLE
PUB-647 - Adding of Subscribers

### PR DESCRIPTION
**Pull request by ChrisS1512**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/PUB-647

### Change description ###

This PR covers the creation of new subscribers. 

It adds the subscriber into two places: 

- Azure Active Directory (Excluding 'Title' - there is no explicit field)
- Azure Table Service

It requires a number of environment variables to be set. Please see docs in ticket for details.

The endpoint that has been added is /account/add, which takes in one or many new subscribers.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/demo/config/AzureConfigurationClient.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
+package uk.gov.hmcts.reform.demo.config;
```
Comment (kurt-uk) [inline src/main/java/uk/gov/hmcts/reform/demo/config/AzureConfigurationClient.java:1]: package needs renaming from demo to pip.account.management

File: src/main/java/uk/gov/hmcts/reform/demo/errorhandling/GlobalExceptionHandler.java [hunk 2]
```diff
@@ -17,21 +16,20 @@
...
+     * and returns a 400 bad request error code.
      * @param ex The exception that has been thrown.
-     * @param request The request made to the endpoint.
      * @return The error response, modelled using the ExceptionResponse object.
      */
-    @ExceptionHandler(AccountNotFoundException.class)
-    public ResponseEntity<ExceptionResponse> handleAccountNotFound(
-        AccountNotFoundException ex, WebRequest request) {
+    @ExceptionHandler(ConstraintViolationException.class)
+    public ResponseEntity<ExceptionResponse> handleMethodValidationError(
```
Comment (kurt-uk) [inline src/main/java/uk/gov/hmcts/reform/demo/errorhandling/GlobalExceptionHandler.java:25]: can we call this method `handle` and then overload the method with the variations of the exceptions that need handling?

**Review by kurt-uk — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/demo/config/AzureConfigurationClient.java (hunk 1 above)
Comment (ChrisS1512) [reply src/main/java/uk/gov/hmcts/reform/demo/config/AzureConfigurationClient.java]: Good spot - I've renamed this now

**Review by marcogagliardi — APPROVED** [review]
👌🏽

**Review by commitBlob — APPROVED** [review]

**Pull request merged**